### PR TITLE
Introduce notify_push in master

### LINKED ISSUE #1809: Introduce notify_push in master
Talked with @jospoortvliet today and we should try to introduce https://github.com/nextcloud/notify_push

It's a daemon, so instead of spawning every 30 minutes or so for a new notification, it' happens instantly.

cc @szaimen

### PR DESCRIPTION
**Pull request by szaimen**
fix #1809
Signed-off-by: szaimen <szaimen@e.mail.de>

### CODE REVIEW

**Comment by szaimen** [comment]
WIP: 
This needs automatic tests (see TODO at the end of the script)
and needs to be tested throughout after the notify_push app gets release to the appsstore.

File: lib.sh [hunk 1]
```diff
@@ -176,7 +176,7 @@ turn_install() {
     JANUS_API_KEY=$(gen_passwd "$SHUF" "a-zA-Z0-9@#*=")
     NC_SECRET=$(gen_passwd "$SHUF" "a-zA-Z0-9@#*=")
     SIGNALING_SERVER_CONF=/etc/signaling/server.conf
-    NONO_PORTS=(22 25 53 80 443 1024 3012 3306 5178 5179 5432 7983 8983 10000 8081 8443 9443)
+    NONO_PORTS=(22 25 53 80 443 1024 3012 3306 5178 5179 5432 7867 7983 8983 10000 8081 8443 9443)
```
Comment (enoch85) [inline lib.sh:179]: I'm glad you remembered! :+1: :+1:

**Comment by szaimen** [comment]
If I understand `notify:setup push_url` correctly, it already runs a selftest directly and should return 1 if something doesn't work.
Otherwise we should be able to run the test manually by using `notify_push:self-test`

**Comment by szaimen** [comment]
this PR needs 
- [ ] https://github.com/nextcloud/notify_push/issues/6
for beeing able to continue

**Comment by szaimen** [comment]
This was successfully tested but still needs the following things:
- [ ] a stable release of notify_push in the appstore
- [ ] #1825 since redis doesn't work currently on all instances
- [ ] remove the todo lines

if all three are done, is this ready

**Comment by enoch85** [comment]
NC 21 was just released. I'm installing a new test VM now.

**Comment by szaimen** [comment]
> NC 21 was just released. I'm installing a new test VM now.

yes but a stable notify_push is still missing -> you will need to uncomment lines that are marked with TODO to make it work since the installation from the appstore will otherwise fail.

**Comment by enoch85** [comment]
We can wait with this. I'd rather merge when everything is in place without workarounds.

**Comment by szaimen** [comment]
Now that the new NC21 release has also focused on the new "High performance backend for files", It is a bid sad, that no stable release of notify_push is available, yet, so that we aren't able to continue with this PR.

**Comment by szaimen** [comment]
And shall we rename the feature to "High Performance Backend for Nextcloud files"?

**Comment by icewind1991** [comment]
Did a stable release

**Comment by szaimen** [comment]
> Did a stable release, sorry for the mistiming with the server release

No problem! Thanks for taking care! :)

**Comment by szaimen** [comment]
This is finally ready and was successfully tested :)

**Pull request merged**